### ISSUE TITLE
Prevent scale from turning off

### ISSUE THREAD

**Issue by mnort9** [erjiang/usbscale#13]
The DYMO M10 has an auto shutdown feature that is triggered after ~3 minutes of inactivity. The only actions that seem to keep the scale awake are changing the weight on the scale and switching the lb/kg mode. 

Any hacks for keeping the scale awake? Or switching the mode via usb?

**Comment by erjiang** [comment]
Unfortunately I do not have a DYMO M10 or other scale with auto-shutdown, so I have no way of even trying out ideas. But I'll leave this issue here in case someone is interested in tackling it.

**Comment by crishoj** [comment]
Has any of you been able to prevent the M10 scale from shutting off automatically?

**Comment by mnort9** [comment]
No. I thought about hard wiring it to toggle the lbs/ounces setting periodically to keep it awake. But that's definitely a hack and won't be reliable...

**Comment by sanmai** [comment]
I have exactly one but why you're bothered with the auto-shutdown?

**Comment by crishoj** [comment]
@sanmai, in our use case the scale is used to weigh books, sometimes frequently, and at other times with 5 minutes of idle time in between each book. Having to check the power status of the scale and turn it on is rather troublesome for the operator.

If no software solution exists, we might modify the scale to detect the power state, and trigger the power switch when the "off" state is detected.

**Comment by Geesu** [comment]
I'd also be curious if this is possible, I've tried to simply write random data to the scale via USB but this doesn't help sadly.

**Comment by Maxhodges** [comment]
@mnort9 pinging the units mode to keep-alive is not a bad idea. It's been done by others
https://stackoverflow.com/questions/36465337/pyusb-stop-dymo-scales-from-shutting-down-automatically

**Comment by zerohimself** [comment]
I just found this discussion while dealing with my brand new dymo m10 scale. I am going to insert a micro controller to pull the scales "display" button's line high or low two times every minute or so. This will have the effect of toggling from oz to g and back each time. That seems to be enough to keep the scale awake, and does not seem to affect the USB readings. I think this is a fairly stupid feature for a scale. I hate to go conspiracy theory on anything, but I have a suspicion that it's an inconvenience to persuade the user to buy a more expensive scale.
